test/scopes_test.js

```diff
@@ -99,3 +100,98 @@ suite('scopeMatch', function() {
     mktest(['foo:bar'], [], false));
 });
 
+suite('scopeIntersection', () => {
+  const fn = utils.scopeIntersection;
+
+  test('single exact match, [string]', () => {
+    const scope = ['foo:bar'];
+    const res = fn(scope, scope);
+
+    assert(isEqual(res, scope), `expected ${scope}`);
+  });
+
+  test('empty [string] in scopesets', () => {
+    const res = fn(['foo:bar'], ['']);
+
+    assert(res.length === 0, 'expected an empty set');
+
+  });
+
+  test('prefix', () => {
+    const scope = ['foo:bar'];
+    const res = fn(['foo:*'], scope);
+
+    assert(isEqual(res, scope), `expected ${scope}`);
+  });
+
+  test('star not at end', () => {
+    const scope = ['foo:bar:bing'];
+    const res = fn(['foo:*:bing'], scope);
+
+    assert(isEqual(res, scope), `expected ${scope}`);
+  });
+
+  test('star at beginning', () => {
+    const scope = ['foo:bar'];
+    const res = fn(['*:bar'], scope);
+
+    assert(isEqual(res, scope), `expected ${scope}`);
+  });
+
+  test('prefix with no star', () => {
+    const scope = ['foo:'];
+    const res = fn(['foo:bar'], scope);
+
+    assert(isEqual(res, scope), `expected ${scope}`);
```

Comment:
Hm, this isn't right.. `foo:` and `foo:bar` are two different scopes, so the intersection should be empty.

Reply:
Don't they share the `foo`?

Reply:
No, prefixes are only relevant with `*`.